lib/csu/mips/crt1_c.c

```diff
@@ -68,6 +82,37 @@ __start(char **ap,
 	argc = * (long *) ap;
 	argv = ap + 1;
 	env  = ap + 2 + argc;
+
+#ifdef SHOULD_PROCESS_CAP_RELOCS
```

Comment:
I wonder how close we would be to a single crt1_c.c shared across most architectures.  I should perhaps look at that upstream I guess.

Reply:
Yeah I was thinking about that, should be pretty straightforward. Especially CHERI architectures should all be sharing a common crt1 in common-cheri...